rules/integrations/azure/credential_access_azure_key_vault_retrieval_from_rare_identity.toml

```diff
@@ -0,0 +1,141 @@
+
+### Response and remediation
+- Investigate the user principal making the excessive retrieval requests to determine if they are authorized to access the Key Vault and its contents. If the user is not authorized, take appropriate actions to block their access and prevent further unauthorized retrievals.
+- Review the application or service making the requests to ensure it is legitimate and authorized to access the Key Vault. If the application is unauthorized or suspicious, consider blocking it and revoking its permissions to access the Key Vault.
+- Assess the impact of the excessive retrieval operations on the Key Vault and its contents. Determine if any sensitive data was accessed or compromised during the retrievals.
+- Implement additional monitoring and alerting for the Key Vault to detect any further suspicious activity or unauthorized access attempts.
+- Consider implementing stricter access controls or policies for Key Vaults to limit excessive retrievals and ensure that only authorized users and applications can access sensitive keys and secrets.
+- Educate users and administrators about the risks associated with excessive retrievals from Key Vaults and encourage them to follow best practices for managing keys and secrets in Azure environments.
+"""
+references = ["https://www.inversecos.com/2022/05/detection-and-compromise-azure-key.html"]
+risk_score = 43
+rule_id = "75c53838-5dcd-11f0-829c-f661ea17fbcd"
+setup = """#### Required Azure Key Vault Diagnostic Logs
+
+To ensure this rule functions correctly, the following diagnostic logs must be enabled for Azure Key Vault:
+- AuditEvent: This log captures all read and write operations performed on the Key Vault, including secret, key, and certificate retrievals. These logs should be streamed to the Event Hub used for the Azure integration configuration.
+"""
+severity = "medium"
+tags = [
+    "Domain: Cloud",
+    "Domain: Storage",
+    "Domain: Identity",
+    "Data Source: Azure",
+    "Data Source: Azure Platform Logs",
+    "Data Source: Azure Key Vault",
+    "Use Case: Threat Detection",
+    "Use Case: Identity and Access Audit",
+    "Tactic: Credential Access",
+    "Resources: Investigation Guide",
+]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+event.dataset : "azure.platformlogs" and
+event.outcome: "success" and
+event.action : (
+    "VaultGet" or
+    "KeyGet" or
+    "KeyList" or
+    "KeyListVersions" or
+    "KeyGetDeleted" or
+    "KeyListDeleted" or
+    "SecretGet" or
+    "SecretList" or
+    "SecretListVersions" or
+    "SecretGetDeleted" or
+    "SecretListDeleted" or
+    "CertificateGet" or
+    "CertificateList" or
+    "CertificateListVersions" or
+    "CertificateGetDeleted" or
+    "CertificateListDeleted" or
+    "CertificatePolicyGet" or
+    "CertificateContactsGet" or
+    "CertificateIssuerGet" or
+    "CertificateIssuersList"
+) and azure.platformlogs.identity.claim.upn: *
```

Comment:
if there is any field that show the target vault/cred owner will be interesting to add a detection (another one) that cover when azure.platformlogs.identity.claim.upn != creds_owner

Reply:
Thats a great question. The correlation - as far as I know - would have to be at the time of creation for that vault key to see whom created and is the original author. There is no flag in Azure or the data that says whom is the creator/owner. Modern access relies on RBAC for users, roles, groups, etc. So it's inherited authorization to access specific keys, secrets, etc.

On that note, we can add the specific vault accessed with `azure.resource.name` to maybe help with signal/noise ratio. So if the UPN has not been seen typically accessing keys/secrets/certs from this specific vault in 14 days, flag.

Reply:
Adjusted to `azure.platformlogs.properties.id`. If that is too noisy, we will reduce to `azure.resource.name`